src/pages/Search/__test__/Search.test.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { Route, MemoryRouter } from 'react-router-dom';
+import {
+  render,
+  cleanup,
+  waitForElementToBeRemoved,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { ThemeProvider } from 'styled-components';
+import axiosMock from 'axios';
+import Search from '../index';
+import Theme from '../../../styles/theme';
+import posts from './posts.json';
+
+jest.mock('axios');
+afterEach(cleanup);
+
+const renderWithRouter = (component, subreddit) => ({
+  ...render(
+    <ThemeProvider theme={Theme}>
+      <MemoryRouter initialEntries={[`/search/${subreddit}`]}>
+        <Route path="/search/:subreddit">{component}</Route>
+      </MemoryRouter>
+    </ThemeProvider>,
+  ),
+});
+
+it('Should show javascript on search input', async () => {
+  const { getByTestId } = renderWithRouter(<Search />, 'javascript');
+
+  const input = getByTestId('search-input');
+
+  expect(input.value).toBe('javascript');
+
+  await waitForElementToBeRemoved(() => getByTestId('loader'));
+});
+
+it('Should load data and show heatmap', async () => {
```

Comment:
It would be great if you would check the numbers in the heatmap boxes as well. That way you would test that the convertData logic is working correctly. You can do that by using the [findAllBy* query](https://testing-library.com/docs/dom-testing-library/api-queries#findallby) and a data-testid on each box.

Reply:
Yeah but results gonna be different if someone else runs tests, because of timezone

Reply:
Ah ok. I guess you might find a way to mock out the timezone. If you can't find a way let me know then I'll give it a try

Reply:
I found a way but need to pass `NODE_ENV === test` and check-in convertedData.js to change  `const date = dayjs.unix(post.created_utc);` to `dayjs.unix(post.created_utc).utc(); `
i dont know if there is better way without changeing `convertedData.js`

Reply:
That's already a good step in the right direction I guess. But we can probably improve it so we don't need the NODE_ENV.

Did you see [this issue](https://github.com/iamkun/dayjs/issues/702)? There they say that you might be able to use the [UTCOffset function](https://day.js.org/docs/en/manipulate/utc-offset#__docusaurus). So I guess what they mean is that you set in your test (for example in  beforeAll) the UTC offset to let's say 0. Then you'll have the same time on all machines during the tests, but not in production

Reply:
I did change UTC and when I go to `convertDate `and `console.log(new Date().getTimezoneOffset())` i get 0 but results still same

Reply:
now everything work I used `timezone-mock` package, I will try to find a way without installing new package